### PR TITLE
semi-dynamic regularization

### PR DESCRIPTION
**Pull request by geoffroyleconte**
This PR is to use the dynamic regularization only for one side of the matrix to factorize.

### CODE REVIEW

**Comment by dpo** [comment]
@tmigot Is this what you had in mind?

File: test/runtests.jl [hunk 1]
```diff
@@ -234,7 +234,37 @@ end
...
+  ϵ = sqrt(eps(eltype(A)))
+  M = spzeros(20, 20)
+  M[1:10, 1:10] = -A * A' - ϵ*I
+  M[11:20, 11:20] = A * A'
+  # M = [-A*A'    0
+  #        0     A*A'] where A*A' is symmetric positive semidefinite
+  b = M * ones(20)
+  x = copy(b)
+  S = ldl_analyze(Symmetric(triu(M), :U))
+  S.r1 = zero(eltype(A))
```
Comment (tmigot) [inline test/runtests.jl:264]: I think a better test would be `S.r1=0`, `S.r2<0` as the case =0, >0 was already possible before. For instance flipping the signs in your example:
```
M[1:10, 1:10] = A * A' + ϵ*I
M[11:20, 11:20] = - A * A'
...
S.r1 = zero(eltype(A))
 S.r2 = -ϵ
```

**Review by tmigot — APPROVED** [review]
Great, thank you!

File: test/runtests.jl (hunk 1 above)
Comment (geoffroyleconte) [reply test/runtests.jl:264]: Yes you are right thank you!

**Comment by dpo** [comment]
Thank you both!

**Pull request merged**